rero_ils/modules/deduplications/api.py

```diff
@@ -0,0 +1,523 @@
+
+        # first author should match
+        candidates = self.check_responsibility_statement(data, candidates)
+
+        candidates = self.check_edition_statement(data, candidates)
+
+        # editor should correspond
+        candidates = self.check_editor(data, candidates)
+
+        # scores are always one if the candidate exists (exact match)
+        common_score = dict(value=1.0)
+        return [
+            (
+                hit._source.pid,
+                hit._source.to_dict(),
+                1.0,
+                dict(
+                    main_type=common_score,
+                    identifier=common_score,
+                    responsibility_statement=common_score,
+                    provision_activity=common_score,
+                    publication_date=common_score,
+                ),
+            )
+            for hit in candidates
+        ]
+
+    @classmethod
+    def normalize(cls, value):
+        """Normalize a given text value.
+
+        1. map each UTF-8 to the closest ascii value
+        2. transform to lower case
+        3. remove extra spaces
+        :param value: str - a given string.
+        :returns: the normalized given value
+        """
+        return re.sub(r"\s+", " ", unidecode(value).lower().strip())
+
+    @classmethod
+    def edit_distance(cls, text1, text2):
+        """Compute a similarity score between two strings.
+
+        Strings are normalized and the score will be normalized between 0 an 1.
+        A score < 0.6 is replaced by 0.
+
+        :param text1: str - first string value.
+        :param text2: str - second string value.
+        :returns: [0.0-1.0] - the score: 1 means a perfect match
+        """
+        score = jaro_winkler(text1, text2, score_cutoff=0.6, processor=cls.normalize)
+        return score
+
+    @classmethod
+    def get_title_score(cls, data, candidate):
+        """Get the title score.
+
+        :returns: the edit distance of the title value.
+        """
+        return cls.edit_distance(
+            main_title_text(data["title"])[0]["_text"],
+            main_title_text(candidate["title"])[0]["_text"],
+        )
+
+    @classmethod
+    def get_series_statement_score(cls, data, candidate):
+        """Get the provision activity score.
+
+        :returns: 1 if both exist and are equal matched, 0 otherwise.
+        """
+        data_field = data.get("seriesStatement", [{"_text": [{"value": ""}]}])
+        c_field = candidate.get("seriesStatement", [{"_text": [{"value": ""}]}])
+        if not data.get("seriesStatement") and not candidate.get("seriesStatement"):
+            return None
+        return (
+            1
+            if cls.normalize(data_field[0]["_text"][0]["value"])
+            == cls.normalize(c_field[0]["_text"][0]["value"])
+            else 0
+        )
+
+    @classmethod
+    def get_edition_statement_score(cls, data, candidate):
+        """Get the edition statement score.
```

Comment:
Compare and score the similarity of the first editionStatement's `_text` in both documents.